src/container.cpp

```diff
@@ -669,21 +667,22 @@ ItemVector Container::getItems(bool recursive /*= false*/)
 			containerItems.push_back(*it);
 		}
 	} else {
-		for (Item* item : itemlist) {
+		for (const auto& item : itemList) {
 			containerItems.push_back(item);
 		}
 	}
 	return containerItems;
 }
 
-Thing* Container::getThing(size_t index) const { return getItemByIndex(index); }
+std::shared_ptr<Thing> Container::getThing(size_t index) const { return getItemByIndex(index); }
 
-void Container::postAddNotification(Thing* thing, const Cylinder* oldParent, int32_t index, cylinderlink_t)
+void Container::postAddNotification(std::shared_ptr<Thing> thing, std::shared_ptr<const Thing> oldParent, int32_t index,
+                                    cylinderlink_t)
 {
-	Cylinder* topParent = getTopParent();
+	auto topParent = getTopParent();
```

Comment:
Could this be `const auto`
```suggestion
	const auto topParent = getTopParent();
```